### PR TITLE
Update content_type collection with content unit types of available p…

### PR DESCRIPTION
**Pull request by goosemania**
…lugins

closes #1642
https://pulp.plan.io/issues/1642

### CODE REVIEW

File: server/test/unit/plugins/loader/test_api.py [hunk 1]
```diff
@@ -93,14 +93,12 @@ def test_check_content_definitions_nothing_old(self, mock_type_definition, mock_
         mock_type_definition.return_value = fake_type
         type_definition = TypeDefinition('steve_holt', 'STEVE HOLT!', 'STEVE HOLT!', 'STEVE HOLT!',
                                          'STEVE HOLT!', 'STEVE HOLT!')
```
Comment (asmacdo) [inline server/test/unit/plugins/loader/test_api.py]: Oh, @jeremycline

File: server/pulp/plugins/loader/api.py [hunk 2]
```diff
@@ -418,10 +418,12 @@ def load_content_types(types_dir=_TYPES_DIR, dry_run=False, drop_indices=False):
         _logger.critical(msg % {'p': types_dir})
         raise IOError(msg % {'p': types_dir})
     descriptors = _load_type_descriptors(types_dir)
+    definitions = parser.parse(descriptors)
```
Comment (asmacdo) [inline server/pulp/plugins/loader/api.py]: Could you add some comments here? It looks like these 2 lines are doing the same thing, but I am guessing that the new function works for Mongoengine converted plugins?

File: server/test/unit/plugins/loader/test_api.py (hunk 1 above)
Comment (jeremycline) [inline server/test/unit/plugins/loader/test_api.py]: How does this git blame to me?

File: server/pulp/plugins/loader/api.py [hunk 3]
```diff
@@ -484,14 +485,35 @@ def _load_type_descriptors(path):
     return descriptors
 
 
-def _load_type_definitions(descriptors, drop_indices=False):
+def _load_type_definitions(definitions, drop_indices=False):
```
Comment (asmacdo) [inline server/pulp/plugins/loader/api.py]: I know this isn't your code, but it doesn't look to me like this function is actually loading anything. If in your work you figured out the role this plays, could you add some comments?

File: server/pulp/plugins/loader/api.py [hunk 4]
```diff
@@ -418,10 +418,12 @@ def load_content_types(types_dir=_TYPES_DIR, dry_run=False, drop_indices=False):
         _logger.critical(msg % {'p': types_dir})
         raise IOError(msg % {'p': types_dir})
     descriptors = _load_type_descriptors(types_dir)
```
Comment (asmacdo) [inline server/pulp/plugins/loader/api.py]: Is this line still being used?

**Comment by asmacdo** [comment]
@goosemania, this looks like good work, but I am a little unclear what is happening here.

From this comment: https://pulp.plan.io/issues/1642#note-5 I expect that all of the available types should be found by your new function `_generate_plugin_definitions()`. Does `parser.parse(descriptors)` catch anything?

File: server/test/unit/plugins/loader/test_api.py (hunk 1 above)
Comment (rbarlow) [inline server/test/unit/plugins/loader/test_api.py]: Austin wrote:
> Oh, @jeremycline <https://github.com/jeremycline>

lol

File: server/pulp/plugins/loader/api.py (hunk 4 above)
Comment (goosemania) [inline server/pulp/plugins/loader/api.py]: yes, the old way (loading json files) is used for node.json only

File: server/pulp/plugins/loader/api.py (hunk 3 above)
Comment (goosemania) [inline server/pulp/plugins/loader/api.py]: I added comments and then decided to remove this one-line function ;) so thank you for your question)

**Comment by goosemania** [comment]
@asmacdo, I've added comments, re-review, please.

File: server/pulp/plugins/loader/api.py [hunk 5]
```diff
@@ -411,17 +411,36 @@ def get_cataloger_by_id(catloger_id):
 
 def load_content_types(types_dir=_TYPES_DIR, dry_run=False, drop_indices=False):
     """
-    :type types_dir: str
+    Check or update database with content unit types information.
+
+    :param types_dir: path to content unit type JSON files,
+                      currently used only for node.json
+    :type types_dir:  str
```
Comment (asmacdo) [inline server/pulp/plugins/loader/api.py]: Personal preference thing (do whatever you want, this is not a blocker for merging) but I prefer putting two spaces between type and the variable. This way the variable names line up and it is easier for me to parse visually.

**Comment by goosemania** [comment]
ok test

**Pull request closed without merging**